.vib/zipkin/runtime-parameters.yaml

```diff
@@ -0,0 +1,477 @@
+                  #!/bin/bash
+
+                  set -o errexit
+                  set -o nounset
+                  set -o pipefail
+
+                  # Set the endpoint URL
+                  host=zipkin
+                  port={{ .Values.service.ports.http }}
+
+                  retry_while() {
+                    local -r cmd="${1:?cmd is missing}"
+                    local -r retries="${2:-12}"
+                    local -r sleep_time="${3:-5}"
+                    local return_value=1
+
+                    read -r -a command <<< "$cmd"
+                    for ((i = 1 ; i <= retries ; i+=1 )); do
+                        "${command[@]}" && return_value=0 && break
+                        sleep "$sleep_time"
+                    done
+                    return $return_value
+                  }
+
+                  zipkin_ready() {
+                      # Test the TCP connection with a timeout
+                      if timeout 5 bash -c "</dev/tcp/$host/$port"; then
+                          return 0
+                      else
+                          return 1
+                      fi
+                  }
+
+                  echo "Waiting for the Zipkin instance"
+                  if ! retry_while "zipkin_ready" 300 2; then
+                    echo "Could not connect to the Zipkin instance"
+                    exit 1
+                  else
+                    echo "Zipkin ready! Running job"
+                    echo "0" > /tmp/ready
+                    curl -k -X POST {{ ternary "https" "http" .Values.tls.enabled }}://zipkin:$port/api/v2/spans -H'Content-Type: application/json' -d @/sampledata/yelp.json
+                  fi
+              startupProbe:
+                exec:
+                  command:
+                    - sh
+                    - -c
+                    - |
+                      if [ $(cat /tmp/ready) = "1" ]; then
+                        exit 0
+                      else
+                        exit 1
+                      fi
+                initialDelaySeconds: 40
+                periodSeconds: 20
+                timeoutSeconds: 1
+                failureThreshold: 15
+                successThreshold: 1
+              securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.containerSecurityContext "context" $) | nindent 12 }}
+              volumeMounts:
+                - name: tmp
+                  mountPath: /tmp
+                  subPath: tmp-dir
+                - name: sample-data
+                  mountPath: /sampledata
+          volumes:
+            - name: tmp
+              emptyDir: {}
+            - name: sample-data
+              configMap:
+                name: zipkin-test-data
+  # ConfigMap with the sample data
+  - |
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: zipkin-test-data
+    data:
+      # Taken from https://github.com/openzipkin/zipkin/blob/master/zipkin-lens/testdata/yelp.json
```

Comment:
I'd drop the ConfigMap and download the `yelp.json` in the job given the container has `curl`. This way, you avoid syncing changes from upstream and simplify the values.

Reply:
That does not work in airgapped environments